app/src/main/java/eu/faircode/email/Core.java

```diff
@@ -1197,9 +1210,11 @@ else if (response.isNO() || response.isBAD())
 
             // Delete local messages not at remote
             Log.i(folder.name + " delete=" + uids.size());
-            for (Long uid : uids) {
-                int count = db.message().deleteMessage(folder.id, uid);
-                Log.i(folder.name + " delete local uid=" + uid + " count=" + count);
+            for (int i = 0; i < uids.size(); i += SYNC_BATCH_SIZE) {
+                int to = Math.min(uids.size(), i + SYNC_BATCH_SIZE);
+                long[] isub = IntStream.range(i, to).mapToLong(l -> uids.get(l)).toArray();
```

Comment:
This requires Java8 which I am trying to avoid for now for maximum compatibility.

Reply:
Compatibility with what? AFAIK AS/R8 will automatically backport bytecode based on your minSdk.

Reply:
Source code compatibility